File: api/src/mappers/message.rs
use crate::dtos::message::MessageResponse;
use crate::models::message::Message;
use uuid::Uuid;

pub fn to_message_response(message: Message) -> MessageResponse {
    MessageResponse {
        message_id: message.message_id,
        conversation_id: message.conversation_id,
        content: message.content,
        sender_id: message.sender_id,
        created_at: message.created_at,
    }
}

File: Frontend/src/types/message.ts
export interface Message {
  message_id: string;
  content: string;
  message_from: string;
  message_to: string;
  created_at: string;
}

export type MessageResponse = Message;

export interface CreateMessage {
  content: string;
  message_to: string;
}

export interface MarkMessageRead {
  user_id: string;
  message_id: string;
}
